lib/python/picongpu/plugins/plot_mpl/transition_radiation_visualizer.py

```diff
@@ -0,0 +1,275 @@
+                        at angles theta and phi over the frequency omega
+                    'sliceovertheta' - shows angular distribution of
+                    transition radiation
+                        at a fixed angle phi and frequency omega
+                    'sliceoverphi' - shows angular distribution of
+                    transition radiation
+                        at a fixed angle theta and frequency omega
+                    'heatmap' - shows angular distribution as heatmap over
+                    both observation angles
+            phi: int
+                index of polar angle for a fixed value
+            theta: int
+                index of azimuth angle for a fixed value
+            omega: int
+                index of frequency for a fixed value, pointless in a spectrum
+        """
+        self.type = kwargs["type"]
+        super().visualize(**kwargs)
+
+    def adjust_plot(self, **kwargs):
+        """
+        Implementation of base class function.
+        """
+        species = kwargs["species"]
+        if self.type == "spectrum":
+            self.ax.set_xlabel(r"Frequency $\omega$ [1/s]")
+            self.ax.set_ylabel(
+                r"Spectral Power $d^2 W / d\omega d\Omega$ [Js]")
+            self.ax.set_xscale("log")
+            self.ax.set_yscale("log")
+            self.ax.set_title("Transition Radiation Spectrum for " + species)
+
+            # care for beautiful figure top axes
+            # activate top axes as a log axes as clone from bottom axes
+            axtop = self.ax.twiny()
+            axtop.set_xscale("log")
+            axtop.autoscale(False)
+            axtop.set_xlim(self.ax.get_xlim())
+
+            # necessary functions for calculation from lambda to omega and
+            # vice versa
+            def calc_lambda(_omega):
+                return 2 * np.pi * const.c / _omega
+
+            def calc_omega(_lambda):
+                return 2 * np.pi * const.c / _lambda
+
+            # calculate limits for top ticks
+            lambda_min = np.ceil(np.log10(calc_lambda(self.ax.get_xlim()[1])))
+            lambda_max = np.floor(np.log10(calc_lambda(self.ax.get_xlim()[0])))
+            # calculate tick locations
+            lambda_locations = np.logspace(lambda_min, lambda_max,
+                                           np.abs(lambda_max - lambda_min) + 1)
+            # set tick locations of top axes
+            axtop.set_xticks(calc_omega(lambda_locations))
+
+            # calculate positions for minor and major ticks for the beauty
+            # of the plot
+            nmaj = 200
+            nmin = 10
+            minorlocs = np.linspace(0.1, 1, nmin, endpoint=True)
+            majorlocs = np.linspace(99, -100, nmaj, endpoint=True)
+            alllocs = np.zeros(nmaj * nmin)
+            for i in range(nmaj):
+                for j in range(nmin):
+                    alllocs[i * nmin + j] = minorlocs[j] * 10 ** majorlocs[i]
```

Comment:
Just as a remark:
The above code looks geared to your current application. 
I would expect issues with the ticks if you go to much larger or much smaller frequency ranges?

Reply:
This is true. But it should work with all frequencies which can are measured in experiments, so I don't think it is a major problem.